cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -1662,7 +1662,348 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
 
         Opcode::Imul => {
             let ty = ty.unwrap();
-            if ty == types::I64X2 {
+
+            // First check for ext_mul_* instructions. Where possible ext_mul_* lowerings
+            // are based on optimized lowerings here: https://github.com/WebAssembly/simd/pull/376
+            if let Some(swiden0_high) = matches_input(ctx, inputs[0], Opcode::SwidenHigh) {
+                if let Some(swiden1_high) = matches_input(ctx, inputs[1], Opcode::SwidenHigh) {
+                    let swiden_input = &[
+                        InsnInput {
+                            insn: swiden0_high,
+                            input: 0,
+                        },
+                        InsnInput {
+                            insn: swiden1_high,
+                            input: 0,
+                        },
+                    ];
```

Comment:
Not sure why `swiden_input` is needed: couldn't the `InsnInput`s be created inline?